packages/cli/src/models/initializer/truffle/Truffle.ts

```diff
@@ -92,23 +97,17 @@ const Truffle = {
       .find(node => node.contractKind === 'library' && node.name === contract.contractName);
   },
 
-  // This function fixes a truffle issue related to HDWalletProvider that occurs when assigning
-  // the network provider as a function (that returns an HDWalletProvider instance) instead of
-  // assigning the HDWalletProvider instance directly.
-  // (see https://github.com/trufflesuite/truffle-hdwallet-provider/issues/65)
-  _setNonceTrackerIfNeeded({ provider }: any): any {
-    const { engine } = provider;
-    if (engine && engine.constructor.name === 'Web3ProviderEngine') {
-      const NonceSubprovider = require('web3-provider-engine/subproviders/nonce-tracker');
-      const nonceTracker = new NonceSubprovider();
-      engine._providers.forEach((aProvider, index) => {
-        if (aProvider.constructor.name === 'ProviderSubprovider') {
-          nonceTracker.setEngine(engine);
-          engine._providers.splice(index, 0, nonceTracker);
-        }
-      });
+  async _checkHdWalletProviderVersion(provider: any, path: string = process.cwd()): Promise<void> {
+    if (provider.constructor.name !== 'HDWalletProvider') return;
+    const packagesList = await npm.list(path);
+    const hdwalletProviderPackage = packagesList
+      .find(packageNameAndVersion => packageNameAndVersion.match(/^truffle-hdwallet-provider@/));
+    if (hdwalletProviderPackage) {
+      const [name, version] = hdwalletProviderPackage.split('@');
```

Comment:
What happens if the package belongs to an organization, such as `@resolver-engine/imports-fs`? Wouldn't the split return an empty string for `name`?

Reply:
Resolved offline by @jcarpanelli - since we are matching on a specific package, doesn't matter